tests/vmcheck/test.sh

```diff
@@ -106,6 +108,7 @@ for tf in $(find . -name 'test-*.sh' | sort); do
     fi
 
     # go back to the original vmcheck deployment if needed
+    if test -z "${VMCHECK_DEBUG:-}"; then
```

Comment:
We should only skip this if we're leaving the loop. Each new test assumes that it's running off a the original commit.  In that workflow, I suppose you're only troubleshooting a single test specified with `TESTS`, so maybe just add a `break` down there to enforce that?